packages/core/test/snapshot.test.js

```diff
@@ -837,14 +837,62 @@ describe('Snapshot', () => {
     }]);
 
     expect(logger.stderr).toEqual([
-      '[percy] Received a duplicate snapshot, ' +
+      '[percy] Received a duplicate snapshot within same test case, ' +
         'the previous snapshot was aborted: /foobar'
     ]);
     expect(logger.stdout).toEqual([
       '[percy] Snapshot taken: /foobar'
     ]);
   });
 
+  it('handles duplicate snapshots when same testCase is passed', async () => {
+    await percy.snapshot([{
+      url: 'http://localhost:8000/foobar',
+      testCase: 'test-case-1',
+      domSnapshot: '<p>Test 1</p>'
+    }, {
+      url: 'http://localhost:8000/foobar',
+      testCase: 'test-case-1',
+      domSnapshot: '<p>Test 2</p>'
+    }]);
+
+    expect(logger.stderr).toEqual([
+      '[percy] Received a duplicate snapshot within same test case, ' +
+        'the previous snapshot was aborted: /foobar'
+    ]);
+    expect(logger.stdout).toEqual([
+      '[percy] Snapshot taken: /foobar'
+    ]);
+  });
+
+  it('handles duplicate snapshots with different test cases', async () => {
+    await percy.snapshot([{
+      url: 'http://localhost:8000/one',
+      testCase: 'test-case-1',
+      domSnapshot: testDOM
+    }, {
+      url: 'http://localhost:8000/one',
+      testCase: 'test-case-2',
+      dom_snapshot: testDOM
+    }, {
+      url: 'http://localhost:8000/one',
+      testCase: 'test-case-3',
+      'dom-snapshot': testDOM
+    }, {
+      url: 'http://localhost:8000/one',
+      testCase: 'test-case-4',
+      domSnapshot: JSON.stringify({ html: testDOM })
+    }]);
+
+    expect(logger.stderr).toEqual([]);
+    expect(logger.stdout).toEqual([
+      '[percy] Snapshot taken: /one',
```

Comment:
we need to update logging and `meta` to include testcase name

Reply:
updated meta as well as logging.